tests/accuracy_utils.py

```diff
@@ -80,26 +75,30 @@ def to_reference(inp, upcast=False):
     if inp is None:
         return None
     ref_inp = inp
-    if TO_CPU:
-        ref_inp = ref_inp.to("cpu")
     if upcast:
         ref_inp = ref_inp.to(torch.float64)
```

Comment:
what about integer? we are not supposed to upcast integers to floats in all occasions.

Reply:
> what about integer? we are not supposed to upcast integers to floats in all occasions.

Now to_reference is only for float inputs.